src/usr/local/sbin/openvpn.attributes.sh

```diff
@@ -18,16 +18,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+lockfile="/tmp/ovpn_${dev}_${username}_${trusted_port}.lock"
+rulesfile="/tmp/ovpn_${dev}_${username}_${trusted_port}.rules"
+anchorname="openvpn/${dev}_${username}_${trusted_port}"
+
 if [ "$script_type" = "client-connect" ]; then
-	if [ -f /tmp/$common_name ]; then
-		/bin/cat /tmp/$common_name > $1
-		/bin/rm /tmp/$common_name
-	fi
+    while [ -f "${lockfile}" ];
+    do
+        sleep 1
+    done
+    touch "${lockfile}"
+
+    cat "${rulesfile}" | sed "s/{clientip}/${ifconfig_pool_remote_ip}/g" > "${rulesfile}.tmp" && mv "${rulesfile}.tmp" "${rulesfile}"
+    /sbin/pfctl -a "openvpn/${dev}_${username}_${trusted_port}" -f "${rulesfile}"
+    rm "${rulesfile}"
+
+    if [ -f /tmp/$common_name ]; then
+        /bin/cat /tmp/$common_name > $1
+        /bin/rm /tmp/$common_name
+    fi
+
+    rm "${lockfile}"
 elif [ "$script_type" = "client-disconnect" ]; then
-	command="/sbin/pfctl -a 'openvpn/$common_name' -F rules"
-	eval $command
-	/sbin/pfctl -k $ifconfig_pool_remote_ip
-	/sbin/pfctl -K $ifconfig_pool_remote_ip
+    while [ -f "${lockfile}" ];
+    do
+        sleep 1
```

Comment:
There should be some kind of alternate escape plan for this loop in case another process gets stuck.

Reply:
Do you have any kind of suggestion other than just adding a timeout? I wasn't too crazy about the way this was implemented but couldn't find any other viable/better solution.

Reply:
Not really. Pick a time (like 10s) and then count that many sleeps and if it's still locked, exit/die

Reply:
Alright. Hoping to have this cleared up by Friday.

Thanks :+1: